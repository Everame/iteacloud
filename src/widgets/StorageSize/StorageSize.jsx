import React from 'react'
import "./styles/StorageSize.scss"

export default function StorageSize({size, maxSize}) {
    const mb = (size/1024).toFixed(3);
    const procent = mb / maxSize * 100;
  return (
    <div className="sizeWrapper">
        <div className="valuesRow">
            <div className="value">
                <h4>{mb}mb</h4>
            </div>
            <div className="value">
                <h4>{maxSize}mb</h4>
            </div>
        </div>
        <div className="progressBar">
            <div className="innerRow" style={{width: `${procent}%`}}></div>
        </div>
    </div>
  )
}
