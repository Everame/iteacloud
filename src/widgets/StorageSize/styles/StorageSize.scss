@media all and (max-width: 480px) {
    .sizeWrapper {
        bottom: calc(var(--index) * 3.5) !important;
    }
}
.sizeWrapper {
    position: fixed;
    left: 3%;
    bottom: calc(var(--index) * 1.3);
    width: 94%;
    height: calc(var(--index) * 1.4);
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-direction: column;
    .valuesRow {
        height: 37%;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: space-between;
        .value {
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            padding: var(--shortMargin);
        }
    }
    .progressBar {
        width: 100%;
        height: 37%;
        border-radius: var(--shortMargin);
        overflow: hidden;
        border: 1px solid var(--unfocus-color);
        .innerRow {
            height: 100%;
            background-color: orange;
            content: '';
        }
    }
}
