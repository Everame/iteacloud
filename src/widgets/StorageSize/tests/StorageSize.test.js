import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router';
import StorageSize from '../StorageSize';

describe("StorageSize", () => {
    it('StorageSize is render', () => {  
        const storageSize = render(
            <MemoryRouter>
              <StorageSize size={2457} maxSize={3.5}/>
            </MemoryRouter>);
        expect(storageSize).toMatchSnapshot();
    });
})