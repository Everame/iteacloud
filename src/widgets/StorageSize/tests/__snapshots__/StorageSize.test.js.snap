// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorageSize StorageSize is render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sizeWrapper"
      >
        <div
          class="valuesRow"
        >
          <div
            class="value"
          >
            <h4>
              2.399
              mb
            </h4>
          </div>
          <div
            class="value"
          >
            <h4>
              3.5
              mb
            </h4>
          </div>
        </div>
        <div
          class="progressBar"
        >
          <div
            class="innerRow"
            style="width: 68.54285714285714%;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sizeWrapper"
    >
      <div
        class="valuesRow"
      >
        <div
          class="value"
        >
          <h4>
            2.399
            mb
          </h4>
        </div>
        <div
          class="value"
        >
          <h4>
            3.5
            mb
          </h4>
        </div>
      </div>
      <div
        class="progressBar"
      >
        <div
          class="innerRow"
          style="width: 68.54285714285714%;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
